### PR TITLE
fix collect tests packs

### PR DESCRIPTION
**Pull request by gal-berger**
## Status
- [] In Progress
- [X] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/27489

## Description
This fix collect packs that are needed for testing a modified pack, even when they belong to another pack.

## Minimum version of Demisto
- [ ] 5.0.0
- [X] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [X] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Review by yuvalbenshalom — CHANGES_REQUESTED** [review]
looks good.
please add unit tests.

File: Tests/scripts/infrastructure_tests/test_collect_tests_and_content_packs.py [hunk 1]
```diff
@@ -183,7 +189,7 @@ def test_changed_runnable_test__unmocked_get_modified_files(self):
         filterd_tests, content_packs = get_mock_test_list(git_diff_ret=self.GIT_DIFF_RET)
 
         assert filterd_tests == {self.TEST_ID}
-        assert content_packs == {"Base", "DeveloperTools", "CommonPlaybooks"}
+        assert content_packs == {"Base", "DeveloperTools", "CommonPlaybooks", "FakePack"}
```
Comment (DeanArbel) [inline Tests/scripts/infrastructure_tests/test_collect_tests_and_content_packs.py:192]: @gal-berger why does "FakePack" appear following your change?

File: Tests/scripts/collect_tests_and_content_packs.py [hunk 2]
```diff
@@ -1161,6 +1171,9 @@ def get_test_list_and_content_packs_to_install(files_string, branch_name, two_be
             tests.add('TestCommonPython')  # test with no integration configured
             tests.add('HelloWorld-Test')  # test with integration configured
             packs_to_install.add("HelloWorld")
+    else:
```
Comment (DeanArbel) [inline Tests/scripts/collect_tests_and_content_packs.py:1174]: I think this should be in the end of `get_test_list_and_content_packs_to_install` along with the call to `get_packs_of_tested_integrations`

File: Tests/scripts/collect_tests_and_content_packs.py [hunk 3]
```diff
@@ -1161,6 +1171,9 @@ def get_test_list_and_content_packs_to_install(files_string, branch_name, two_be
...
+        packs_of_collected_tests = conf.get_packs_of_collected_tests(tests, id_set)
+        packs_to_install = packs_to_install.union(packs_of_collected_tests)
```
Comment (DeanArbel) [inline Tests/scripts/collect_tests_and_content_packs.py:1176]: I think this should be in the end of `get_test_list_and_content_packs_to_install` along with the call to `get_packs_of_tested_integrations`

Comment (gal-berger) [reply Tests/scripts/collect_tests_and_content_packs.py]: you're right, I've changed it.

**Review by DeanArbel — APPROVED** [review]
Good job :)

**Review by yuvalbenshalom — APPROVED** [review]

**Pull request merged**